### PR TITLE
[Merged by Bors] - fix: export ast field info for <local>.<field>

### PR DESCRIPTION
**Pull request by dselsam**
This commit addresses https://github.com/leanprover-community/mathport/issues/58#issuecomment-1007316139 Note that the title and first comment in that issue are unrelated and were addressed by [61b4659b7bcbf2c61b09db8370487b6f78403701](https://github.com/leanprover-community/mathport/commit/61b4659b7bcbf2c61b09db8370487b6f78403701)

### CODE REVIEW

File: src/frontends/lean/parser.cpp [hunk 1]
```diff
@@ -2092,21 +2123,28 @@ optional<expr> parser::resolve_local(name const & id, pos_info const & p, list<n
...
 
     if (allow_field_notation && !id.is_atomic() && id.is_string()) {
-        if (auto r = resolve_local(id.get_prefix(), p, extra_locals)) {
+        ast_data prefix_data = id_data; // TODO(dselsam): implement
+        name n = id_data.m_value = id.get_prefix();
+        if (auto s = resolve_local(n, id_data, p, extra_locals)) {
             auto field_pos = p;
             field_pos.second += id.get_prefix().utf8_size();
-            return some_expr(save_pos(mk_field_notation_compact(*r, id.get_string()), field_pos));
+            return some_expr(save_pos(mk_field_notation_compact(*this, *s, field_pos, id.get_string()), field_pos));
```
Comment (digama0) [inline src/frontends/lean/parser.cpp:2147]: this calls `save_pos` twice

File: src/frontends/lean/parser.cpp [hunk 2]
```diff
@@ -2092,21 +2123,28 @@ optional<expr> parser::resolve_local(name const & id, pos_info const & p, list<n
...
     if (auto it1 = m_local_decls.find(id)) {
-        return some_expr(copy_with_new_pos(*it1, p));
+        expr r = copy_with_new_pos(*it1, p);
+        finalize_ast(id_data.m_id, r);
+        return some_expr(r);
     }
 
     if (allow_field_notation && !id.is_atomic() && id.is_string()) {
-        if (auto r = resolve_local(id.get_prefix(), p, extra_locals)) {
+        ast_data prefix_data = id_data; // TODO(dselsam): implement
```
[the remark this answers is not in the document]
Comment (dselsam) [reply src/frontends/lean/parser.cpp]: good catch -- stale line, removed

File: src/frontends/lean/parser.cpp (hunk 1 above)
Comment (dselsam) [reply src/frontends/lean/parser.cpp]: good catch -- removed the save_pos at that line

**Review by digama0 — APPROVED** [review]

**Comment by gebner** [comment]
bors merge

**Comment by gebner** [comment]
bors merge

**Pull request closed without merging**